### PR TITLE
GSP_GPU::StoreDataCache stubbed

### PR DESCRIPTION
**Pull request by mailwl**
This avoid endless spam of "unimplemented StoreDataCache..." in some games

### CODE REVIEW

File: src/core/hle/service/gsp_gpu.cpp [hunk 1]
```diff
@@ -705,6 +705,33 @@ static void ReleaseRight(Interface* self) {
...
+static void StoreDataCache(Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    u32 address = cmd_buff[1];
+    u32 size = cmd_buff[2];
+    u32 process = cmd_buff[4];
+
+    cmd_buff[0] = IPC::MakeHeader(0x1F, 0x1, 0);
+    cmd_buff[1] = RESULT_SUCCESS.raw; // No error
+
+    LOG_DEBUG(Service_GSP, "(STUBBED) called address=0x%08X, size=0x%08X, process=0x%08X", address,
```
Comment (bunnei) [inline src/core/hle/service/gsp_gpu.cpp]: Stubbed functions are supposed to be WARNING, but then I guess it still spams unless we change the default log level :P

**Comment by yuriks** [comment]
Tbh I'd even be fine wuth removing the STUBBED tag and converting the
log to TRACE. This function is now implemented, it's just a no-op in our
implementation.


--yuriks





On Fri, Jan 13, 2017, at 09:04 AM, bunnei wrote:

> *@bunnei* commented on this pull request.



> 

> In src/core/hle/service/gsp_gpu.cpp[1]:


> > + *      2 : Size + *      3 : Value 0, some descriptor for the
> >   KProcess Handle + *      4 : KProcess handle + *  Outputs: + *
> >   1 : Result of function, 0 on success, otherwise error code + */
> >   +static void StoreDataCache(Interface* self) { +    u32* cmd_buff
> >   = Kernel::GetCommandBuffer(); +    u32 address = cmd_buff[1]; +
> >   u32 size = cmd_buff[2]; +    u32 process = cmd_buff[4]; + +
> >   cmd_buff[0] = IPC::MakeHeader(0x1F, 0x1, 0); +    cmd_buff[1] =
> >   RESULT_SUCCESS.raw; // No error + +    LOG_DEBUG(Service_GSP,
> >   "(STUBBED) called address=0x%08X, size=0x%08X, process=0x%08X",
> >   address,
>
> Stubbed functions are supposed to be WARNING, but then I guess it
> still spams unless we change the default log level :P
> — You are receiving this because you commented. Reply to this email
> directly, view it on GitHub[2], or mute the thread[3].
> 


Links:

  1. https://github.com/citra-emu/citra/pull/2428#pullrequestreview-16606779
  2. https://github.com/citra-emu/citra/pull/2428#pullrequestreview-16606779

File: src/core/hle/service/gsp_gpu.cpp (hunk 1 above)
Comment (yuriks) [inline src/core/hle/service/gsp_gpu.cpp]: TBH I'd even be fine with removing the STUBBED tag and converting the
log to TRACE. This function is now implemented, it's just a no-op in our
implementation.

Comment (mailwl) [inline src/core/hle/service/gsp_gpu.cpp]: all these cache functions additionally checks address ranges, then call appropriate SVC call.
we may fully implement service functions, but stub SVC calls in other PR. nothing changes, but code will be more clean

**Pull request merged**